Formula/g/goplus.rb

```diff
@@ -6,6 +6,14 @@ class Goplus < Formula
   license "Apache-2.0"
   head "https://github.com/goplus/gop.git", branch: "main"
 
+  # Upstream creates releases that use a stable tag (e.g., `v1.2.3`) but are
+  # labeled as "pre-release" on GitHub before the version is released, so it's
+  # necessary to use the `GithubLatest` strategy.
+  livecheck do
+    url :stable
+    strategy :github_latest
```

Comment:
```suggestion
  livecheck do
    url :stable
    regex(/^v?(\d+(?:\.\d+)+)$/i)
```

I may be overlooking something but this doesn't seem like a repository that creates a "pre-release" release with a stable tag and then removes the "pre-release" label when it's stable/released (per the comment). From what I'm seeing, they created a release for `v1.2.0-pre.1` but didn't mark it as "pre-release" until you created an issue, which is a different scenario.

If upstream creates releases for unstable versions like `v1.2.0-pre.1` but they don't reliably mark them as "pre-release", then `GithubLatest` could surface an unstable version (i.e., introducing a different problem).

Using the standard regex for Git tags like `1.2.3`/`v1.2.3` would be sufficient here and we wouldn't have to worry about the GitHub release situation unless they start having a long gap between tag and release. With the most recent releases, the release was created less than an hour after the tag (sometimes within minutes), so checking the tags seems fine.

The formula's `stable` URL is a tag archive (not a release asset), so I'm in favor of simply using the regex for now and we can always revisit this (e.g., to use `GithubLatest`) if that doesn't end up being sufficient in the long term.

Reply:
> I may be overlooking something but this doesn't seem like a repository that creates a "pre-release" release with a stable tag and then removes the "pre-release" label when it's stable/released (per the comment). From what I'm seeing, they created a release for v1.2.0-pre.1 but didn't mark it as "pre-release" until you created an issue, which is a different scenario.

I dont think upstream knows what they are doing.

Reply:
I shall think we should include this pre-release note if that is what we are doing as general practice.  :)

Reply:
What I'm saying is that I don't see any evidence that upstream creates releases using a stable tag (e.g., `1.2.3`) and marks it as "pre-release" before it's stable/released (removing the "pre-release" status when it's released). That's what the `livecheck` block comment is explaining but that isn't the situation in this repository (unlike the other formulae using this comment).

Looking back at [previous snapshots of the GitHub releases page](https://web.archive.org/web/20221001000000*/https://github.com/goplus/gop/releases), the issue is that upstream creates GitHub releases for unstable versions (e.g., `v1.2.0-pre.1`, `v1.1.0-rc3`, `v1.1.0-alpha3`) without marking them as pre-release. In this scenario, if upstream continues to not mark unstable releases as "pre-release", the "latest" release on GitHub may be an unstable and `GithubLatest` will either A) return an unstable version or B) return an `Unable to get versions` error. Like you, I don't have much faith that upstream will reliably mark unstable releases as "pre-release" (like they've done for `v1.2.0-pre.1`) going forward.

As mentioned above, checking the Git tags using the standard regex 1) avoids unstable versions and 2) won't run into any issues with upstream's poor irregular process. I agree that upstream isn't careful about releases but the solution is to check the Git tags using a stricter regex (to avoid the `v1.2.0-pre.1` tag). `GithubLatest` isn't appropriate in this context and it's likely to lead to further issues rather than fixing this.

Reply:
Now, I see the point wiht the `Unable to get versions` error issue. let me remove that comment then (also copy the comments into the upstream so that they aware of this downstream impact. )